rest/changes_api_test.go

```diff
@@ -198,6 +198,117 @@ func TestDocDeletionFromChannel(t *testing.T) {
+
+	rt1 := RestTester{SyncFn: `function(doc) {channel(doc.channel)}`}
+	defer rt1.Close()
+
+	a := rt1.ServerContext().Database("db").Authenticator()
+
+	// Create user in rt1:
+	alice, _ := a.NewUser("alice", "letmein", channels.SetOf("zero"))
+	a.Save(alice)
+
+	rt2 := RestTester{RestTesterBucket: rt1.RestTesterBucket, SyncFn: `function(doc) {channel(doc.channel)}`}
+	rt2.RestTesterServerContext = NewServerContext(&ServerConfig{
+		Facebook:       &FacebookConfig{},
+		AdminInterface: &DefaultAdminInterface,
+	})
+
+	server := base.UnitTestUrl()
+	bucketName := rt1.RestTesterBucket.GetName()
+
+	_, err := rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
+		BucketConfig: BucketConfig{
+			Server: &server,
+			Bucket: &bucketName},
+		Name: "db",
+	})
+
+	assertNoError(t, err, "Failed to add database to rest tester")
+
+	defer rt2.Close()
+
+	// Create a doc Alice can see:
+	response := rt1.Send(request("PUT", "/db/alpha", `{"channel":"zero"}`))
+
+	// Check the _changes feed:
+	rt1.ServerContext().Database("db").WaitForPendingChanges()
+	var changes struct {
+		Results []db.ChangeEntry
+	}
+	response = rt1.Send(requestByUser("GET", "/db/_changes", "", "alice"))
+	log.Printf("_changes looks like: %s", response.Body.Bytes())
+	json.Unmarshal(response.Body.Bytes(), &changes)
+	assert.Equals(t, len(changes.Results), 1)
+	since := changes.Results[0].Seq
+	assert.Equals(t, since.Seq, uint64(1))
+
+	assert.Equals(t, changes.Results[0].ID, "alpha")
+	rev1 := changes.Results[0].Changes[0]["rev"]
+
+	// Delete the document:
+	assertStatus(t, rt1.Send(request("DELETE", "/db/alpha?rev="+rev1, "")), 200)
+
+	// Get the updates from the _changes feed:
+	rt1.WaitForPendingChanges()
+	response = rt1.Send(requestByUser("GET", fmt.Sprintf("/db/_changes?since=%s", since),
+		"", "alice"))
+	log.Printf("_changes looks like: %s", response.Body.Bytes())
+	changes.Results = nil
+	json.Unmarshal(response.Body.Bytes(), &changes)
+	assert.Equals(t, len(changes.Results), 1)
+
+	assert.Equals(t, changes.Results[0].ID, "alpha")
+	assert.Equals(t, changes.Results[0].Deleted, true)
+	assert.DeepEquals(t, changes.Results[0].Removed, base.SetOf("zero"))
+	rev2 := changes.Results[0].Changes[0]["rev"]
+
+	// Now get the deleted revision:
+	response = rt1.Send(requestByUser("GET", "/db/alpha?rev="+rev2, "", "alice"))
+	assert.Equals(t, response.Code, 200)
+	log.Printf("Deletion looks like: %s", response.Body.Bytes())
+	var docBody db.Body
+	json.Unmarshal(response.Body.Bytes(), &docBody)
+	assert.DeepEquals(t, docBody, db.Body{"_id": "alpha", "_rev": rev2, "_deleted": true})
+
+	// Access without deletion revID shouldn't be allowed (since doc is not in Alice's channels):
+	response = rt1.Send(requestByUser("GET", "/db/alpha", "", "alice"))
+	assert.Equals(t, response.Code, 403)
+
+	// A bogus rev ID should return a 404:
+	response = rt1.Send(requestByUser("GET", "/db/alpha?rev=bogus", "", "alice"))
+	assert.Equals(t, response.Code, 404)
```

Comment:
Do we need to be explicit about the response on these checks? I.e. 404-missing vs 404-deleted?